spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -318,8 +325,30 @@ def check_spyder_kernel_callback(self, reply):
         # Process kernel reply
         data = reply.get('data')
         if data is not None and 'text/plain' in data:
-            is_spyder_kernel = data['text/plain']
-            if 'SpyderKernel' in is_spyder_kernel:
+            version = ast.literal_eval(data['text/plain'])
+            if not version:
+                # The running_under_pytest() part can be removed when
+                # spyder-kernels 3 makes it into conda. This is needed for
+                # the test_conda_env_activation test
+                if running_under_pytest():
+                    return
+
+                if self.is_spyder_kernel:
+                    self.ipyclient.show_kernel_error(
+                        _("spyder_kernels version must be updated"))
+                return
+
+            if not check_version_range(version, SPYDER_KERNELS_VERSION):
+                if "dev0" not in version:
+                    # Development versions are acceptable
+                    self.ipyclient.show_kernel_error(
+                        ERROR_SPYDER_KERNEL_VERSION.format(
+                            version,
+                            SPYDER_KERNELS_VERSION
+                            ))
```

Comment:
Given that the kernel version is going to be validated like this now, I think we should remove the code we have for that in:

https://github.com/spyder-ide/spyder/blob/03fde9d4131b8c32874248900d46435c1f9eec03/spyder/plugins/ipythonconsole/widgets/main_widget.py#L1480-L1514

and replace the message shown here by the one in that code, adding to it also the actual Spyder-kernels version because we have a simple way to detect it.